cnf-tests/testsuites/e2esuite/fec/fec.go

```diff
@@ -0,0 +1,169 @@
+package fec
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	sriovfecv1 "github.com/open-ness/openshift-operator/sriov-fec/api/v1"
+
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/discovery"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/nodes"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/utils"
+)
+
+var (
+	nodeName string
+	numVfs   = 2
+)
+
+var _ = Describe("fec", func() {
+	BeforeEach(func() {
+		if discovery.Enabled() {
+			// TODO: change this to read the sriovFecClusterConfig and validate all the nodes
+			Skip("sriov-fec test doesn't support discovery mode")
+		}
+
+		err := cleanSriovFecClusterObject()
+		Expect(err).ToNot(HaveOccurred())
+
+		err = createSriovFecClusterObject()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		err := cleanSriovFecClusterObject()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("Expose resource on the node", func() {
+		It("should show resources under the node", func() {
+			Eventually(func() int64 {
+				testedNode, err := client.Client.Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				resNum, _ := testedNode.Status.Allocatable[corev1.ResourceName(utils.Acc100ResourceName)]
+				allocatable, _ := resNum.AsInt64()
+				return allocatable
+			}, 10*time.Minute, time.Second).Should(Equal(int64(numVfs)))
+		})
+	})
+})
+
+func cleanSriovFecClusterObject() error {
+	sriovFecCluster := &sriovfecv1.SriovFecClusterConfig{}
+	err := client.Client.Get(context.TODO(), runtimeClient.ObjectKey{Name: utils.SriovFecClusterConfigName, Namespace: utils.SriovFecOperatorNamespace}, sriovFecCluster)
+	if errors.IsNotFound(err) {
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return client.Client.Delete(context.TODO(), sriovFecCluster)
+}
+
+func createSriovFecClusterObject() error {
+	nodesWithAcc100, err := getSriovFecNodes()
```

Comment:
Can you split this in "findNodeWithFec" or something?
Here we set the global as a side effect, and it's not clear where that is coming from.
I'd prefer the global to be set in beforeEach and to use it as a parameter for this

Reply:
I'd prefer (but feel free to leave it as it is) to have this take the node as a parameter, so wrapping all the node selection logic in a single function and calling it in beforeEach

Reply:
done I split the function